test/helpers/utils.spec.ts

```diff
@@ -132,4 +174,75 @@ describe('redirectsForNext404Route', () => {
       { force: false, from: '/fr/test', status: 404, to: '/server/pages/fr/404.html' },
     ])
   })
+
+  it('returns static 404 redirects when LEGACY_FALLBACK_FALSE is not set', async () => {
+    const expected = {
+      dynamicRewrites: [
+        {
+          force: false,
+          from: '/_next/data/test/getStaticProps/:id.json',
+          status: 404,
+          to: '/server/pages/404.html',
+        },
+        {
+          force: false,
+          from: '/getStaticProps/:id',
+          status: 404,
+          to: '/server/pages/404.html',
+        },
+      ],
+      dynamicRoutesThatMatchMiddleware: [],
+    }
+
+    expect(generateDynamicRewrites(route)).toStrictEqual(expected)
+  })
+
+  it('does not return static 404 redirects when LEGACY_FALLBACK_FALSE is true', async () => {
```

Comment:
We only want to ensure that `LEGACY_FALLBACK_FALSE` has a value set. It doesn't matter what the value is.

```suggestion
  it('does not return static 404 redirects when LEGACY_FALLBACK_FALSE is set', async () => {
```